scripts/build_component_container.sh

```diff
@@ -4,35 +4,52 @@ set -e;
 
 source ./scripts/util.sh
 
+# Sanity check for not arguments being passed.
 if [ "$#" -eq 0 ]
 then
-    util::error "No directory provided to build"
+    util::error "No arguments provided to build. Expected two."
     exit 1
 fi
 
-executable=$(basename $(dirname "${1}"))
+if [ -z "$1" ]
+then
+    util::error "No directory argument provided to build."
+    exit 1
+fi
+
+if [ -z "$2" ]
+then
+    util::error "No build architecture argument provided to build."
+    exit 1
+fi
+
+dir_name="$1"
+build_architecture="$2"
+
+executable=$(basename $(dirname ${dir_name}))
 
-echo "${1}" | grep -Eq ^components/producers/.*$ && executable="${executable}-parser" || true
-echo "${1}" | grep -Eq ^components/enrichers/.*$ && executable="${executable}" || true
-echo "${1}" | grep -Eq ^components/consumers/.*$ && executable="${executable}" || true
+echo ${dir_name} | grep -Eq ^components/producers/.*$ && executable="${executable}-parser" || true
+echo ${dir_name} | grep -Eq ^components/enrichers/.*$ && executable="${executable}" || true
+echo ${dir_name} | grep -Eq ^components/consumers/.*$ && executable="${executable}" || true
 
-executable_src_path=$(dirname "${1}")
-executable_path=$(dirname "${1}")/"${executable}"
+executable_src_path=$(dirname $dir_name)
+executable_path=$(dirname $dir_name)/${build_architecture}/"${executable}"
```

Comment:
nit: can you encapsulate please the BASH variables in curly brackets here and here: https://github.com/ocurity/dracon/pull/343/commits/eea40403dc9292ecd571b7b09613b41b1615a2f2#diff-e886250d151c99b2b701ddeab8e0e58bcac8f434e02ead4aafaab12462fa8e86R25 for consistency?